src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -63,14 +63,23 @@ public interface IHostEnvironment : IChannelProvider, IProgressChannelProvider
         IHost Register(string name, int? seed = null, bool? verbose = null);
 
         /// <summary>
-        /// Flag which indicate should we stop any code execution in this host.
+        /// The catalog of loadable components (<see cref="LoadableClassAttribute"/>) that are available in this host.
         /// </summary>
-        bool IsCancelled { get; }
+        ComponentCatalog ComponentCatalog { get; }
+    }
 
+    [BestFriend]
+    internal interface ICancelableHost
```

Comment:
>ICancelableHost [](start = 23, length = 15)

Shouldn;t it inherit from IHost? (for example, it should include Rand)

Reply:
No, it shouldn't.